### ISSUE TITLE
Questions regarding the accuracy evaluation 

### ISSUE THREAD

**Issue by zhangshuoneu** [jiah-cloud/Align3R#12]
This is a very meaningful work, and I appreciate the open-sourcing of the code, which has made a significant contribution to the field of dynamic scene reconstruction. After reading the paper and the code, I have a few questions regarding the accuracy evaluation that I would like to ask.

1. Were all the accuracy evaluation experiments conducted using the RTX 4090 GPU? When the number of images caused OOM (Out of Memory), was a hierarchical optimization approach used for evaluation?

2. Why is the number of images used for evaluating depth and pose metrics inconsistent? I noticed that for depth evaluation, Bonn used 110 frames and TUM dynamics used 50 frames. However, for pose evaluation, both Bonn and TUM dynamics used 30 frames, and Bonn was only evaluated on 5 datasets.

3. In the pose evaluation section, the paper mentions, "For DROID-SLAM, DPVO, Robust-CVD, and CasualSAM, we use the results reported in MonST3R." Given that the number of images used differs, this approach seems somewhat unfair.

**Comment by zhangshuoneu** [comment]
Additionally, I have noticed that the optical flow model used in the open-source code differs from the one in MonST3R. Could you elaborate on the reasons for changing the optical flow model? Furthermore, does the replacement of the optical flow model contribute positively to the accuracy metrics?

**Comment by jiah-cloud** [comment]
> This is a very meaningful work, and I appreciate the open-sourcing of the code, which has made a significant contribution to the field of dynamic scene reconstruction. After reading the paper and the code, I have a few questions regarding the accuracy evaluation that I would like to ask.
> 
> 1. Were all the accuracy evaluation experiments conducted using the RTX 4090 GPU? When the number of images caused OOM (Out of Memory), was a hierarchical optimization approach used for evaluation?
> 2. Why is the number of images used for evaluating depth and pose metrics inconsistent? I noticed that for depth evaluation, Bonn used 110 frames and TUM dynamics used 50 frames. However, for pose evaluation, both Bonn and TUM dynamics used 30 frames, and Bonn was only evaluated on 5 datasets.
> 3. In the pose evaluation section, the paper mentions, "For DROID-SLAM, DPVO, Robust-CVD, and CasualSAM, we use the results reported in MonST3R." Given that the number of images used differs, this approach seems somewhat unfair.

<html>
<body>


<p>Thank you very much for your attention to our work.</p>
<ol>
<li>
<p>Yes, all the experiments were conducted on the same RTX 4090 GPU. When the number of images caused Out of Memory (OOM), we employed the hierarchical optimization strategy to address the issue.</p>
</li>
<li>
<p>The reason is straightforward—it helps save evaluation time and GPU memory. Additionally, we tested different subsets of samples to reflect the generalization ability across varying distributions to some extent. Regarding why Bonn was only evaluated on five datasets, we followed prior works such as <strong>DepthCrafter</strong> and <strong>MonST3R</strong> for consistency.</p>
</li>
<li>
<p>Under the camera pose setting, our primary focus was on comparing our method with similar approaches like <strong>DUSt3R</strong> and <strong>MonST3R</strong>. For methods like <strong>DROID-SLAM</strong>, <strong>DPVO</strong>, <strong>Robust-CVD</strong>, and <strong>CasualSAM</strong>, our intention was to provide a reasonable reference point. This table design approach has been adopted in many other works, such as <strong>Real-time Photorealistic Dynamic Scene Representation and Rendering with 4D Gaussian Splatting (ICLR 2024)</strong> and <strong>Spacetime Gaussian Feature Splatting for Real-Time Dynamic View Synthesis (CVPR 2024)</strong>.</p>
<p>Of course, to ensure a fairer comparison, we evaluated results under the same settings as <strong>MonST3R</strong> (90 frames in TUM-dynamics):</p>
</li>
</ol>

Dataset | Method | ATE (m) | RPE Trans (deg) | RPE Rot (deg)
-- | -- | -- | -- | --
TUM-dynamics | MonST3R | 0.063 | 0.009 | 1.217
  | Ours (Depth Pro) | 0.027 | 0.018 | 0.446


<p>For <strong>Sintel</strong>, the settings used in the main text are identical to those of <strong>MonST3R</strong>. Thanks a lot for your reminder, we will also update the above results in our newest version.</p>

**Comment by jiah-cloud** [comment]
> Additionally, I have noticed that the optical flow model used in the open-source code differs from the one in MonST3R. Could you elaborate on the reasons for changing the optical flow model? Furthermore, does the replacement of the optical flow model contribute positively to the accuracy metrics?

We found that MonST3R used different optical flow models for different datasets. For simplicity, we standardized our approach by using a single model and selected the most appropriate pre-trained weights (provided by the official code). As discussed in Supplementary Section 8, we observed that the flow loss significantly contributes to camera pose estimation but has a relatively smaller impact on depth estimation. Therefore, selecting an appropriate optical flow model positively influences accuracy metrics. You may also consider trying some of the latest state-of-the-art optical flow models.

**Comment by zhangshuoneu** [comment]
Congratulations on your paper accepted by CVPR, that's fantastic! Regarding the accuracy evaluation, I have a detail I'd like to ask about. For the Bonn dataset, is the number of frames used for camera pose evaluation 30:60? And for the TUM Dynamics dataset, is it consistent with MonST3R, taking 30 frames from the first 90 frames with a stride of 3?

**Comment by jiah-cloud** [comment]
> Congratulations on your paper accepted by CVPR, that's fantastic! Regarding the accuracy evaluation, I have a detail I'd like to ask about. For the Bonn dataset, is the number of frames used for camera pose evaluation 30:60? And for the TUM Dynamics dataset, is it consistent with MonST3R, taking 30 frames from the first 90 frames with a stride of 3?

Yes, you are right
